src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1371,119 +2869,158 @@ void test_syscollector_dbsync_network_address_valid_msg_modified(void **state)
+
+static const char *valid_empty_queries[] = { "agent 001 netinfo save NULL|NULL|"
+                "NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|0|NULL|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|1|NULL|"
+                "NULL|NULL" };
+
+void test_syscollector_netinfo_valid_groups_empty (void **state)
+{
+    Eventinfo *lf = *state;
+    const char *result = "ok ";
+    int sock = 1;
+    size_t count = sizeof(valid_empty_queries) / sizeof(*valid_empty_queries);
+
+    for (int i = 0; i < count; i++)
+    {
+        expect_any(__wrap_wdbc_query_ex, *sock);
+        expect_string(__wrap_wdbc_query_ex, query, valid_empty_queries[i]);
+        expect_any(__wrap_wdbc_query_ex, len);
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
+
+    int ret = DecodeSyscollector(lf, &sock);
+
+    assert_int_not_equal(ret, 0);
+}
+
+static const char *valid_empty_address_array_queries[] = { "agent 001 netinfo save NULL|NULL|"
+                "NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|0|NULL|NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|0|0.0.0.0|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|1|NULL|"
+                "NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|1|0000:0000:0000:0:0000:0000:0000:0000|"
+                "NULL|NULL" };
+
+void test_syscollector_netinfo_valid_address_array_empty (void **state)
 {
     Eventinfo *lf = *state;
+    const char *result = "ok ";
+    int sock = 1;
+    size_t count = sizeof(valid_empty_address_array_queries) / sizeof(*valid_empty_address_array_queries);
+
+    for (int i = 0; i < count; i++)
+    {
+        expect_any(__wrap_wdbc_query_ex, *sock);
+        expect_string(__wrap_wdbc_query_ex, query, valid_empty_address_array_queries[i]);
+        expect_any(__wrap_wdbc_query_ex, len);
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
 
-    const char *query = "agent 001 dbsync packages INSERTED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"format\":\"1\","
-            "\"name\":\"2\","
-            "\"priority\":\"3\","
-            "\"groups\":\"4\","
-            "\"size\":\"5\","
-            "\"vendor\":\"6\","
-            "\"install_time\":\"7\","
-            "\"version\":\"8\","
-            "\"architecture\":\"9\","
-            "\"multiarch\":\"10\","
-            "\"source\":\"11\","
-            "\"description\":\"12\","
-            "\"location\":\"13\","
-            "\"checksum\":\"17\","
-            "\"item_id\":\"18\""
-        "}";
+    int ret = DecodeSyscollector(lf, &sock);
+
+    assert_int_not_equal(ret, 0);
+}
+
+void test_syscollector_netinfo_valid_network_end (void **state)
```

Comment:
Due to the number of functions defined, mainly because of the code coverage, and in the absence of other tools, it seems more appropriate to not continue adding functions excessively.